backend/inspirehep/disambiguation/cli.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import click
+from elasticsearch_dsl import Q
+from inspire_utils.record import get_values_for_schema
+from invenio_db import db
+
+from inspirehep.records.api import AuthorsRecord
+from inspirehep.records.models import RecordsAuthors
+from inspirehep.search.api import AuthorsSearch
+
+
+@click.group()
+def disambiguation():
+    """Commands for disambiguation"""
+
+
+@disambiguation.command()
+def clean_stub_authors():
+    stub_authors_query = Q("term", stub=True)
+    stub_authors = (
+        AuthorsSearch().query(stub_authors_query).source(["control_number"]).execute()
+    )
+    stub_authors_control_numbers = [
+        ("aut", str(author["control_number"])) for author in stub_authors
+    ]
+    stub_authors_db = list(
+        AuthorsRecord.get_records_by_pids(stub_authors_control_numbers)
+    )
+    db.session.connection(execution_options={"isolation_level": "SERIALIZABLE"})
+    for author in stub_authors_db:
+        if not author["stub"]:
+            continue
+        author_bai = get_values_for_schema(author["ids"], "INSPIRE BAI")[0]
+        author_papers = RecordsAuthors.query.filter(
+            RecordsAuthors.id_type == "INSPIRE BAI",
+            RecordsAuthors.author_id == author_bai,
+        ).all()
```

Comment:
* you don't need `all` results here, all you want to know if this is empty or not, so you can do `count` or `first` instead of `all`.
* I think you need to add an index for this, otherwise it will be very slow.